Epsilon/Services/LearningOutcomeCanvasResultService.cs

```diff
@@ -68,12 +69,17 @@ public async IAsyncEnumerable<LearningDomainSubmission> GetSubmissions(string st
 
         await Task.WhenAll(submissionsTask, domainOutcomesTask);
 
-        foreach (var submission in submissionsTask?.Result?.Courses?.SelectMany(static c => c.Submissions.Nodes))
+        if (submissionsTask.Result?.Courses == null)
+        {
+            throw new NoNullAllowedException();
```

Comment:
Use this one and please add a message like such: `throw new NullReferenceException("msg here");`